stock_full_location_reservation/models/stock_move.py

```diff
@@ -72,18 +72,24 @@ def _full_location_reservation_create_move(
                 product, qty, location, package
             )
         )
+        self.ensure_one()
         if self.picking_type_id.merge_move_for_full_location_reservation:
             # To be able to be merged, the new move should use the same source location as
             # the original one.
             new_move.location_id = self.location_id
             self._do_unreserve()
             # Don't merge at confirm
             new_move._action_confirm(merge=False)
-            new_move = (
+            merged_move = (
                 (new_move | self)
                 .with_context(skip_undo_full_location_reservation=True)
                 ._merge_moves(merge_into=self)
             )
+            # If the concerned product is not the one of the original move,
+            # it hasn't been merged.
+            if len(merged_move) > 1:
+                return new_move
+            return merged_move
```

Comment:
Would it not easier to check the product before hand?
If it is a different product we do not need to merge the move, since the _merge_moves gets as arg merge_into so the candidate list is only the self. 
`if self.product_id == product and self.picking_type_id.merge_move_for_full_location_reservation`

Reply:
I thought about it but wanted to let the keys evaluation for merge do its job and not doing the job twice

Reply:
But it would simplify the code
```
        self.ensure_one()
        if self.product_id == product and self.picking_type_id.merge_move_for_full_location_reservation:
            # To be able to be merged, the new move should use the same source location as
            # the original one.
            new_move.location_id = self.location_id
            self._do_unreserve()
            # Don't merge at confirm
            new_move._action_confirm(merge=False)
            return = (
                (new_move | self)
                .with_context(skip_undo_full_location_reservation=True)
                ._merge_moves(merge_into=self)
            )
```

Reply:
Would it not also be possible, to only change the product_uom_qty?
Because nothing more is happening in the merge_move. 
```

    def _full_location_reservation_create_move(
        self, product, qty, location, package=None
    ):
        if self.product_id == product and self.picking_type_id.merge_move_for_full_location_reservation:
            self._do_unreserve()
            self.product_uom_qty += qty
            return self
        new_move = self.copy(
            default=self._full_location_reservation_prepare_move_vals(
                product, qty, location, package
            )
        )
```